runtime/interpreter/variable.go

```diff
@@ -19,11 +19,26 @@
 package interpreter
 
 type Variable struct {
-	Value Value
+	value  Value
+	getter func() Value
 }
 
-func NewVariable(value Value) *Variable {
+func (v *Variable) GetValue() Value {
+	if v.getter != nil {
+		v.value = v.getter()
```

Comment:
I have two questions:

When would the value be loaded for a contract?

Will this be thread-safe? As in, will there be a situation where two or more programs trying to load a 'contract' at the same time (not sure if that's a valid use-case though)?

Reply:
The interpreter used to eagerly load the contract value right when the interpreter interpreted a contract declaration: it would load the value and declare a variable with the value.

Now, when the interpreter interprets a variable declaration, the interpreter only declares a variable with a getter function, and the first time the variable's value is accessed, the getter function is called, and the contract value is loaded, and placed in the variable.

If the contract variable is never accessed, the contract value is never loaded. This is very often the case, so we can safe the unnecessary load.

There is no need to be thread-safe in the interpreter, it is single-threaded and so there is no potential for other parts of interpretation trying to load it at the same time. If external code attempts to e.g. access `Globals` and a variable's value in a concurrent way, then that's undefined behaviour.